### DISCUSSION TITLE
This database failed to load

### DISCUSSION THREAD

**Discussion by wa1gon** [apache/couchdb#3467] [Q&A]
## Description

After the Couchdb server has been idle for a while Fauxton has a message about "This database failed to load."  Sometimes it is a few and other times it is all.

Via the API I get a message that says something about no db shards.

Retrying the request is almost always successful.

## Steps to Reproduce

Not sure how to reproduce it, but bad error.

## Expected Behaviour

No errors.

## Your Environment

All databases are small.  Largest is under 30 meg and only 10 databases.  Only 2 used at any one time.

Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz   3.60 GHz
24 gig memory

[TIP]:  # ( Include as many relevant details about your environment as possible. )
[TIP]:  # ( You can paste the output of curl http://YOUR-COUCHDB:5984/ here. )

* CouchDB version used:
* **{"couchdb":"Welcome","version":"3.1.1","git_sha":"ce596c65d","uuid":"d06eab0052c4773e6fcb9c30ee0c87e4","features":["access-ready","partitioned","pluggable-storage-engines","reshard","scheduler"],"vendor":{"name":"The Apache Software Foundation"}}**
* Browser name and version:
Chrome Version 89.0.4389.90 (Official Build) (64-bit) and Brave Version 1.22.70 Chromium: 89.0.4389.105 (Official Build) (64-bit)
* Operating system and version:
Windows 10 Version 20H2

## Configuration

Section | Option | Value |  
-- | -- | -- | --
admin | -pbkdf2-235b18a3be7b0180f4b8bad5af60290956c50c14,ed76ad8746e84befbbc6ea1e9654e456,10 |  
compressible_types | text/*, application/javascript, application/json, application/xml |  
compression_level | 8 |  
backlog | 512 |  
bind_address | 0.0.0.0 |  
max_db_number_for_dbs_info_req | 100 |  
port | 5984 |  
prefer_minimal | Cache-Control, Content-Length, Content-Range, Content-Type, ETag, Server, Transfer-Encoding, Vary |  
require_valid_user | false |  
server_options | [{recbuf, undefined}] |  
socket_options | [{sndbuf, 262144}, {nodelay, true}] |  
n | 20 |  
q | 2 |  
credentials | false |  
allow_persistent_cookies | true |  
auth_cache_size | 50 |  
authentication_db | _users |  
authentication_redirect | /_utils/session.html |  
iterations | 10 |  
require_valid_user | false |  
secret | 00f791fe71e81f6db39410c7266f1d65 |  
timeout | 5184000 |  
database_prefix | userdb- |  
delete_dbs | false |  
enable | false |  
attachment_stream_buffer_size | 4096 |  
changes_doc_ids_optimization_threshold | 100 |  
database_dir | ./data |  
default_engine | couch |  
default_security | admin_only |  
file_compression | snappy |  
max_dbs_open | 2500 |  
max_document_size | 8000000 |  
os_process_timeout | 5000 |  
users_db_security_editable | false |  
uuid | d06eab0052c4773e6fcb9c30ee0c87e4 |  
view_index_dir | ./data |  
couch | couch_bt_engine |  
enable | true |  
partitioned\|\|* | true |  
allow_jsonp | false |  
authentication_handlers | {couch_httpd_auth, cookie_authentication_handler}, {couch_httpd_auth, default_authentication_handler} |  
bind_address | 127.0.0.1 |  
enable_cors | false |  
enable_xframe_options | false |  
max_http_request_size | 4294967296 |  
port | 5986 |  
secure_rewrites | true |  
socket_options | [{sndbuf, 262144}] |  
couch_mrview | true |  
concurrency | 10 |  
ratio | 0.01 |  
compaction | false |  
os_process | true |  
read | true |  
shard_sync | false |  
view_update | true |  
write | true |  
level | info |  
writer | stderr |  
os_process_limit | 100 |  
reduce_limit | true |  
connection_timeout | 30000 |  
http_connections | 20 |  
interval | 60000 |  
max_churn | 20 |  
max_jobs | 500 |  
retries_per_request | 5 |  
socket_options | [{keepalive, true}, {nodelay, false}] |  
ssl_certificate_max_depth | 3 |  
startup_jitter | 5000 |  
verify_ssl_certificates | false |  
worker_batch_size | 500 |  
worker_processes | 4 |  
port | 6984 |  
algorithm | sequential |  
max_count | 1000 |  
name | The Apache Software Foundation

## Additional Context

[TIP]:  # ( Add any other context about the problem here. )

**Comment by wohali** [comment]
Could this be because of a token timeout? The default `couch_httpd_auth timeout` is 600.

If you run a `curl` command in a loop from the command line, passing your login and password in the URL, and have the loop fail if any one request fails, do you ever see a failure? If not, it's definitely the timeout.

You could try increasing your timeout and see if the problem reduces in frequency. If so... that'd be a smoking gun.

**Reply by wa1gon** [reply]
Not the problem, my couch_httpd_auth is set to 5184000.  Also if that was the problem then Fauxton would do a redirect.

**Reply by wohali** [reply]
What errors, if any, are in your logfile? If you increase logging level to `info`, it might help pinpoint the issue.
